pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -137,18 +138,11 @@ func NewProvider(
 		}
 	}
 
-	var registryMirror *v1alpha1.RegistryMirrorConfiguration
-	if clusterConfig.Spec.RegistryMirrorConfiguration != nil {
-		registryMirror = clusterConfig.Spec.RegistryMirrorConfiguration
-	} else {
-		registryMirror = nil
-	}
-
 	return &Provider{
 		clusterConfig:         clusterConfig,
 		datacenterConfig:      datacenterConfig,
 		machineConfigs:        machineConfigs,
-		stackInstaller:        stack.NewInstaller(docker, writer, helm, constants.EksaSystemNamespace, clusterConfig.Spec.ClusterNetwork.Pods.CidrBlocks[0], registryMirror),
+		stackInstaller:        stack.NewInstaller(docker, writer, helm, constants.EksaSystemNamespace, clusterConfig.Spec.ClusterNetwork.Pods.CidrBlocks[0], registrymirror.FromClusterRegistryMirrorConfiguration(clusterConfig.Spec.RegistryMirrorConfiguration)),
```

Comment:
nit, probably simpler?:
```suggestion
		stackInstaller:        stack.NewInstaller(docker, writer, helm, constants.EksaSystemNamespace, clusterConfig.Spec.ClusterNetwork.Pods.CidrBlocks[0], registrymirror.FromCluster(clusterConfig)),
```